### PR TITLE
gluon-check-connection proposal

### PR DESCRIPTION
**Pull request by CodeFetch**
It's untested and I won't have time in the near future to work on it.
It aims to simplify connectivity tests for other packages like ffho's autoupdater-wifi-fallback, scheduled-domain-switch and the offline-ssid package.

### CODE REVIEW

File: docs/package/gluon-check-connection.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Defining target groups in the site.conf will overwrite existing ones with the same
+name when performing a *sysupgrade* or by triggering *gluon-reconfigure*.
+
+Configuration via UCI
+---------------------
+
+Packages can use gluon-check-connection to be triggered after connection checks.
+For this they can define the following *script* attributes:
+
+script : an entry for defining the ping target
```
Comment (CodeFetch) [inline docs/package/gluon-check-connection.rst]: an entry for configuring and defining the triggered command

File: package/gluon-check-connection/luasrc/usr/bin/gluon-check-connection [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+			if states[group] then
+				state_online = true
+			else
+				state_offline = true
+			end
+		end
+	end
+
+	if script['onchange'] and not state_changed then
+		return
```
Comment (CodeFetch) [inline package/gluon-check-connection/luasrc/usr/bin/gluon-check-connection]: break instead of return

**Comment by blocktrron** [comment]
Thanks for your proposal!

As this PR is not meant to be merged in it's current state I'm closing it, leaving the information there to be picked up by someone in the future.

**Comment by CodeFetch** [comment]
@blocktrron I'm waiting for feedback and someone willing to test it. If you close it, it won't get any intention.

**Comment by mweinelt** [comment]
> It's untested and I won't have time in the near future to work on it.

Sounds very much like it's a code dump without any intention to work on it.

**Comment by CodeFetch** [comment]
It's a "proposal". As you don't comment on the concept itself, I see this as an ACK. So I'm going to test it when I have time. Yes, I don't really have an intention to work on it as I guess it's just fine as it is if it runs.

**Comment by mweinelt** [comment]
We feel strongly that a package that pings out into the network to determine some state it can derive from local information has no place in the gluon core repository.

We are discussing a proposal on IRC right now with @blocktrron @AiyionPrime and @NeoRaider and will see what comes off it.

**Comment by CodeFetch** [comment]
@mweinelt @blocktrron @AiyionPrime @NeoRaider  This package is supposed to be a generic test facility using ping. How can you test whether internet connection is available using "local" information only? If you would have looked at the PR you would have seen that this PR is both about performing local and global tests.

**Comment by mweinelt** [comment]
We simply don't think it's reasonable for networks with 500-1500 nodes to basically ping flood any host on the internet. We think the local state, like a relatively fresh IPv6 default route, or a gateway announcement has sufficient informational value to derive a status for most use cases. 

That will be my last message on the matter.

**Pull request closed without merging**